### PR TITLE
feat: queue

### PR DESCRIPTION
**Pull request by thantos**
Config:

* [x] Fifo: true/false
* [x] Encryption: yes/no
* [x] MessageGroupID

Client-side APIs:

* [x] SendMessage
* [x] SendMessageBatch
* [x] ChangeMessageVisibility
* [x] DeleteMessage
* [x] DeleteMessageBatch

Tasks:

* Core
  * [x] Queue Call
  * [x] Queue Resource + Spec
  * [x] Queue Physical Name Property
  * [x] Queue Handler
* [x] Runtime
   * [x] Queue Client
   * [x] Workers
      * [x] Queue Call Executor
      * [x] Queue Physical Name Retriever
      * [x] Queue Handler Worker
    * [x] Workflow
      * [x] Queue Workflow Events
      * [x] Queue Call Workflow Executor
      * [x] Queue Call Workflow Eventual Factory
      * [x] Wire Up Factory, Executor, and QueueName property
    * [x] Transaction Worker
      * [x] Support new Properties
    * [x] Replay
      * [x] Wire up Properties
* [x] Environment
   * [x] AWS
      * [x] AWS Queue Client
      * [x] Dep Injection
      * [x] AWS Queue Worker
      * [x] Update other workers
    * [x] Local
      * [x] Implement Local Queue Client
      * [x] Local Queue Event
    * [x] Test
 * [x] Infra
    * [x] CDK
       * [x] Queue Service
       * [x] Add Queue Service and things to Queue
       * [x] Add Queue to other Services
 * [x] Tests

### CODE REVIEW

File: packages/@eventual/core-runtime/src/local/clients/queue-client.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  type QueueProvider,
+} from "../../index.js";
+
+export interface QueueRetrieveMessagesRequest {
+  maxMessages?: number;
+  visibilityTimeout?: DurationSchedule;
+}
+
+/**
+ * TODO: implement message deduplication
```
Comment (sam-goodwin) [inline packages/@eventual/core-runtime/src/local/clients/queue-client.ts:24]: Issue or do? Seems straightforward?

Comment (sam-goodwin) [reply packages/@eventual/core-runtime/src/local/clients/queue-client.ts:24]: I guess the hard part is the content based deduplication

File: packages/@eventual/core-runtime/src/handlers/transaction-worker.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export function createTransactionWorker(
   const propertyRetriever = new AllPropertyRetriever({
     BucketPhysicalName: unsupportedPropertyRetriever,
     OpenSearchClient: unsupportedPropertyRetriever,
+    QueuePhysicalName: unsupportedPropertyRetriever,
```
[the remark this answers is not in the document]
Comment (thantos) [reply packages/@eventual/core-runtime/src/handlers/transaction-worker.ts:41]: Same reason as bucket physical name, the current support for these are through the client, which we don't currently inject into the transaction worker. Would need to decouple the physical name calculation (and overrides) from the client first.

File: packages/@eventual/core-runtime/src/local/clients/queue-client.ts (hunk 1 above)
Comment (thantos) [reply packages/@eventual/core-runtime/src/local/clients/queue-client.ts:24]: The issue is the time window, but may not be hard to do. Save all of the values/hashes with a date, separated by group for fifo. Ignore messages with the same value/hash until the time is up.

File: packages/@eventual/compiler/src/ast-util.ts [hunk 3]
```diff
@@ -74,13 +74,33 @@ export function isBucketHandlerMemberCall(call: CallExpression): boolean {
...
-      // bucket.stream(events, ""handlerName", options, async () => { })
+      // bucket.stream(events, "handlerName", options, async () => { })
       return call.arguments.length === 3 || call.arguments.length === 4;
     }
   }
   return false;
 }
 
+/**
+ * A heuristic for identifying a {@link CallExpression} that is a call to an `on` handler.
```
[the remark this answers is not in the document]
Comment (thantos) [reply packages/@eventual/compiler/src/ast-util.ts:85]: ohh, this is changed, forgot to change the infer.

File: packages/@eventual/aws-cdk/src/queue-service.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  queueService: QueueService<any>;
+  queue: QueueRuntime;
+}
+
+export interface IQueue {
+  queue: sqs.Queue;
+  handler: QueueHandler;
+}
+
+class Queue extends Construct implements IQueue {
```
[the remark this answers is not in the document]
Comment (thantos) [reply packages/@eventual/aws-cdk/src/queue-service.ts:123]: IQueue is just properties (above)?

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**